src/game/gameflow.c

```diff
@@ -0,0 +1,669 @@
+
+        case GFE_CUTSCENE:
+            if (type != GFL_SAVED) {
+                const int16_t level = g_CurrentLevel;
+                const int32_t result = Game_Cutscene_Start(ptr[1]);
+                g_CurrentLevel = level;
+                if (result == 2
+                    && (type == GFL_STORY || type == GFL_MIDSTORY)) {
+                    return GFD_EXIT_TO_TITLE;
+                }
+                if (result == 3) {
+                    return GFD_EXIT_GAME;
+                }
+            }
+            ptr += 2;
+            break;
+
+        case GFE_LEVEL_COMPLETE:
+            if (type != GFL_STORY && type != GFL_MIDSTORY) {
+                if (LevelStats(g_CurrentLevel)) {
+                    return GFD_EXIT_TO_TITLE;
+                }
+                option = GFD_START_GAME | (g_CurrentLevel + 1);
+            }
+            ptr++;
+            break;
+
+        case GFE_DEMO_PLAY:
+            if (type != GFL_SAVED && type != GFL_STORY
+                && type != GFL_MIDSTORY) {
+                return Demo_Start(ptr[1]);
+            }
+            ptr += 2;
+            break;
+
+        case GFE_JUMP_TO_SEQ:
+            ptr += 2;
+            break;
+
+        case GFE_SET_TRACK:
+            g_GF_MusicTracks[ntracks] = ptr[1];
+            Game_SetCutsceneTrack(ptr[1]);
+            ntracks++;
+            ptr += 2;
+            break;
+
+        case GFE_SUNSET:
+            if (type != GFL_STORY && type != GFL_MIDSTORY) {
+                g_GF_SunsetEnabled = true;
+            }
+            ptr++;
+            break;
+
+        case GFE_LOADING_PIC:
+            ptr += 2;
+            break;
+
+        case GFE_DEADLY_WATER:
+            if (type != GFL_STORY && type != GFL_MIDSTORY) {
+                g_GF_DeadlyWater = true;
+            }
+            ptr++;
+            break;
+
+        case GFE_REMOVE_WEAPONS:
+            if (type != GFL_STORY && type != GFL_MIDSTORY
+                && type != GFL_SAVED) {
+                g_GF_RemoveWeapons = true;
+            }
+            ptr++;
+            break;
+
+        case GFE_GAME_COMPLETE:
+            DisplayCredits();
+            if (GameStats(g_CurrentLevel)) {
+                return GFD_EXIT_TO_TITLE;
+            }
+            option = GFD_EXIT_TO_TITLE;
+            ptr++;
+            break;
+
+        case GFE_CUT_ANGLE:
+            if (type != GFL_SAVED) {
+                g_CineTargetAngle = ptr[1];
+            }
+            ptr += 2;
+            break;
+
+        case GFE_NO_FLOOR:
+            if (type != GFL_STORY && type != GFL_MIDSTORY) {
+                g_GF_NoFloor = ptr[1];
+            }
+            ptr += 2;
+            break;
+
+        case GFE_ADD_TO_INV:
+            if (type != GFL_STORY && type != GFL_MIDSTORY) {
+                if (ptr[1] < 1000) {
+                    g_GF_SecretInvItems[ptr[1]]++;
+                } else if (type != GFL_SAVED) {
+                    g_GF_Add2InvItems[ptr[1]]++;
```

Comment:
I think this needs to be `g_GF_Add2InvItems[ptr[1] - 1000]++;`. This is used at the start of Tibetan Foothills and Barkhang Monastery to add The Seraph (puzzle4) to the inventory as it's carried over from The Deck. It's also used in HSH to give `key1`.

Reply:
Excellent catch! Fixed: https://github.com/LostArtefacts/TR2X/compare/34dc3b1807bfd929ac08a11feb1da1a0f5476c41..5a182c9b0519b48bd6f66d9de6c88d4f05ed7efd

New test build: https://home.wind.garden/serve/pub/TR2X-0.2-309-g5a182c9-dirty-Windows.zip